### PR TITLE
server: add -r option on startup to specify the initial role

### PR DESCRIPTION
**Pull request by drdstech**
We don't want to affect MySQL services when we upgrade xenon, so after the upgrade need to let the old LEADER continue to run as the LEADER and skip leaderStartShellCommand.

### CODE REVIEW

File: src/raft/raft.go [hunk 1]
```diff
@@ -45,53 +45,57 @@ type RaftMeta struct {
...
+	F                        *Follower
+	I                        *Idle
+	IV                       *Invalid
+	LN                       *Learner
+	peers                    map[string]*Peer // all peers expect SuperIDLE
+	idlePeers                map[string]*Peer // all SuperIDLE peers
+	stats                    model.RaftStats
+	skipPurgeBinlog          bool // if true, purge binlog will skipped
+	skipCheckSemiSync        bool // if true, check semi-sync will skipped
+	semiSyncTimeoutFor2Nodes uint64
```
Comment (andyli029) [inline src/raft/raft.go:78]: It is better to add comment?

**Comment by andyli029** [comment]
Xenon Coverage / Coverage Failed.
@drdstech

**Comment by drdstech** [comment]
push update:  
fix TestRaftLeaderChangeToMasterError failed

File: src/raft/leader.go [hunk 2]
```diff
@@ -427,7 +426,10 @@ func (r *Leader) prepareSettingsAsync() {
 		}
 		r.WARNING("mysql.SetReadWrite.done")
 		r.WARNING("6. start.vip.prepare")
-		if err := r.leaderStartShellCommand(); err != nil {
+		if r.initRole == LEADER {
```
Comment (andyli029) [inline src/raft/leader.go:429]: The step need to discuss.

File: src/raft/raft.go [hunk 3]
```diff
@@ -45,53 +45,57 @@ type RaftMeta struct {
...
-	stats               model.RaftStats
-	skipPurgeBinlog     bool // if true, purge binlog will skipped
-	skipCheckSemiSync   bool // if true, check semi-sync will skipped
-	isBrainSplit        bool // if true, follower can upgrade to candidate
-	gtid                model.GTID
+	log                      *xlog.Log
+	mysql                    *mysql.Mysql
+	cmd                      common.Command
+	conf                     *config.RaftConfig
+	initRole                 State
```
Comment (andyli029) [inline src/raft/raft.go:52]: Add the comment.

**Comment by andyli029** [comment]
Split the pr into several prs.

**Comment by drdstech** [comment]
push update:
1. split pr
2. add doc and comments.

**Comment by andyli029** [comment]
Ping @hustjieke

**Pull request merged**